### PR TITLE
feat: Support KnexJS

### PR DESCRIPTION
**Pull request by mgagliardo91**
This PR adds functionality to support the query interface [KnexJs](http://knexjs.org/).

### CODE REVIEW

File: lib/query-interface/knex.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+      case 'pg':
+        return 'postgres';
+      default:
+        return client;
+    }
+  }
+
+  parseDatabaseError(error) {
+    if (
+      error.message && error.message.indexOf('could not obtain lock') > -1
```
Comment (branchvincent) [inline lib/query-interface/knex.js:25]: so you can't catch an error by it's class in node? or is knex throwing a generic `Error` here?

**Review by branchvincent — APPROVED** [review]

File: lib/query-interface/knex.js (hunk 1 above)
Comment (mgagliardo91) [reply lib/query-interface/knex.js:25]: knex is throwing a generic `Error` yea

File: README.md [hunk 2]
```diff
@@ -56,6 +66,29 @@ importantLock
...
+const KnexJs = require('knex');
+const DistributedLock = require('@ndustrialio/sequelize-distributed-lock');
+
+const knex = new KnexJs({...});
+
+const importantLogic = async () => {
+  // will never run at the same time
+}
+
+const importantLock = new DistributedLock('my-important-logic-lock', {
```
[the remark this answers is not in the document]
Comment (mgagliardo91) [reply README.md:81]: wdym?

[the remark this answers is not in the document]
Comment (branchvincent) [reply README.md:81]: lmao nvm, my brain read `importantLogic` twice

**Pull request merged**